pkg/sql/revert.go

```diff
@@ -94,21 +219,38 @@ func RevertTables(
 				TargetTime:        targetTime,
 				IgnoreGcThreshold: ignoreGCThreshold,
 			})
-		}
-		b.Header.MaxSpanRequestKeys = batchSize
+			// TODO(ssd): We should probably be setting an
+			// admission header here has well.
+			b.Header.MaxSpanRequestKeys = batchSize
 
-		if err := db.Run(ctx, &b); err != nil {
-			return err
-		}
+			log.VEventf(ctx, 2, "RevertRange %s %s", span, targetTime)
+			if err := db.Run(ctx, &b); err != nil {
+				return err
+			}
+
+			if l := len(b.RawResponse().Responses); l != 1 {
+				return errors.AssertionFailedf("expected single response, got %d", l)
+			}
+
+			resp := b.RawResponse().Responses[0].GetRevertRange()
+			if resp == nil {
+				return errors.AssertionFailedf("expected RevertRangeResponse, got: %v", resp)
+			}
+
+			completed := *span
+			span = resp.ResumeSpan
+
+			if resp.ResumeSpan != nil {
+				if !resp.ResumeSpan.Valid() {
```

Comment:
nit: perhaps we add some logging around resume key spans and the number of attempts to revert the original span?

Reply:
We have a VEventf above that can be added on each revert range call.  Note that a resume span is absolutely expected.